### PR TITLE
Remove Xcode 12.5 GitHub Actions Jobs

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

- Remove GitHub Actions jobs using Xcode 12.5 which will no longer be supported in v6.

### Checklist

- ~[ ] Added a changelog entry~

- @scannillo 
- @sarahkoop

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -62,7 +14,7 @@ jobs:
         # Allowing warnings until Xcode 13 CocoaPods vendored frameworks issue is resolved: https://github.com/CocoaPods/CocoaPods/issues/10731
       - name: Run pod lib lint 
         run: pod lib lint --allow-warnings
-  carthage_xcode_beta:
+  carthage:
     name: Carthage (Xcode 13-beta)
```
Comment (sshropshire) [inline .github/workflows/build.yml:18]:
```suggestion
    name: Carthage (Xcode 13)
```
Although Xcode 13 is technically unreleased. Not&nbsp;💯 &nbsp;here.

**Review by sshropshire — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (scannillo) [reply .github/workflows/build.yml:18]: I think we're going to leave the 'beta' tag here so we remember to update it to the GA in September

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -2,54 +2,6 @@ name: Build
...
-      - name: Use Xcode 12.5
-        run: sudo xcode-select -switch /Applications/Xcode_12.5.app
-      - name: Use current branch
-        run: sed -i '' 's/branch = .*/branch = \"'"$GITHUB_HEAD_REF"'\";/' SampleApps/SPMTest/SPMTest.xcodeproj/project.pbxproj
-      - name: Run swift package resolve
-        run: cd SampleApps/SPMTest && swift package resolve
-      - name: Build & archive SPMTest
-        run: xcodebuild -project 'SampleApps/SPMTest/SPMTest.xcodeproj' -scheme 'SPMTest' clean build archive CODE_SIGNING_ALLOWED=NO
-  cocoapods_xcode_beta:
     name: CocoaPods (Xcode 13-beta)
```
Comment (scannillo) [inline .github/workflows/build.yml:5]:
```suggestion
    name: CocoaPods
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (scannillo) [inline .github/workflows/build.yml:18]:
```suggestion
    name: Carthage
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -84,7 +36,7 @@ jobs:
         run: cd SampleApps/CarthageTest && carthage update --use-xcframeworks
       - name: Build CarthageTest
         run: xcodebuild -project 'SampleApps/CarthageTest/CarthageTest.xcodeproj' -scheme 'CarthageTest' clean build CODE_SIGNING_ALLOWED=NO
-  spm_xcode_beta:
+  spm:
     name: SPM (Xcode 13-beta)
```
Comment (scannillo) [inline .github/workflows/build.yml:40]:
```suggestion
    name: SPM
```

**Pull request merged**